notebooks/sentiment_analysis.py

```diff
@@ -0,0 +1,338 @@
+"""
+# Sentiment analysis (using IPUs)
+
+Integration of the Graphcore Intelligence processing unit (IPU) and the HuggingFace transformer library means that it only takes a few lines of code to perform complex tasks which require deep learning.
+
+In this notebook we perform **sentiment analysis**: we use natural language processing models to classify text prompts. 
+We follow [this blog post by Federico Pascual](https://huggingface.co/blog/sentiment-analysis-python) and test 5 different models available on HuggingFace Hub to highlight different model properties of the models that can be leveraged for downstream tasks.
+
+The ease of use of the `pipeline` interface lets us quickly experiment with the pre-trained models and identify which one will work best.
+This simple interface means that it is extremely easy to access the fast inference performance of the IPU on your application.
+
+![Widget inference on a text classification task](images/text_classification.png)
+
+While this notebook is focused on using the model (inference), our [text_classification](text_classification.ipynb) notebook will show you how to fine tune a model for a specific task using the `datasets` package.
+"""
+"""
+First of all, lets make sure your environment has the latest version of [🤗 Optimum Graphcore](https://github.com/huggingface/optimum-graphcore) available.
+"""
+%pip install "optimum-graphcore>=0.4, <0.5"
+%pip install emoji == 0.6.0
+
+%load_ext autoreload
+%autoreload 2
+
+import os
+os.environ["POPTORCH_LOG_LEVEL"] = "ERR"
+
+"""
+## Using transformers pipelines on the IPU
+
+The simplest way to use a model on the IPU is to use the `pipeline` function. It provides a set of models which have been validated to work on a given task, to get started choose the task and call the `pipeline` function:
+"""
+from optimum.graphcore import pipelines
+sentiment_pipeline = pipelines.pipeline("sentiment-analysis")
+
+"""
+Because no model was supplied the pipeline defaulted to `distilbert-base-uncased-finetuned-sst-2-english`. We are helpfully warned that we should provide the `max_length` argument if we were to put this model in production, but while we are experimenting we will leave it as is.
+
+Now it's time to test our first prompts. Let's start with some very easy to classify text:
+"""
+simple_test_data = ["I love you", "I hate you"]
+sentiment_pipeline(simple_test_data)
+
+"""
+Reassuringly, the model got it right! And with a high degree of confidence, more than 99.9% in both cases.
+
+You'll have noticed that this was a bit slow, it took several seconds to provide the answer. This is because of the computing architecture of the IPU, during the first execution of a pipeline, the model needs to be compiled and loaded into the IPU. While this takes some time, these overheads only the first execution of a pipeline! 
```

Comment:
I would rephrase that - no need to say "noticed". Just something like "Compilation of the model happens at the first call, subsequent calls will be faster"

Reply:
yeah that was unnecessary I went with:

> The first call to the pipeline was a bit slow, it took several seconds to provide the answer. This behaviour is due to compilation of the model which happens on the first call.
On subsequent prompts it is much faster: